src/Model/Behavior/UploadBehavior.php

```diff
@@ -99,7 +100,14 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
             $path = $this->getPathProcessor($entity, $data, $field, $settings);
             $basepath = $path->basepath();
             $filename = $path->filename();
-            $data['name'] = $filename;
+            if (is_string($data)) {
+                $temp = [];
+                $temp['name'] = $filename;
+                $temp['data'] = $data;
+                $data = $temp;
```

Comment:
What is the purpose of this?

Reply:
When upload base64 $data is just a string f.ex
`Y2FrZXBocA==`
It's not an array and doesn't have an name.
Could you think about a better solution to this problem?
Here is the same thing https://github.com/FriendsOfCake/cakephp-upload/pull/495/files#diff-84c66ea1e7013949dabde1157612da57R122
$data['size'] and $data['type'] are not set when upload base64, when doing regular upload they are already provided by php.
Maybe implement another 2 interface?

Reply:
Also because of the https://github.com/FriendsOfCake/cakephp-upload/pull/495/files#diff-84c66ea1e7013949dabde1157612da57R122
When uploading base64 we can't use the size and type features to have them set in the database

Reply:
Can you show how this would be used in practice? It's not clear whats going on here.

Reply:
You mean with the 2 extra interfaces?

Reply:
```
            $data = $entity->get($field);
            $path = $this->getPathProcessor($entity, $data, $field, $settings);
            $basepath = $path->basepath();
            $filename = $path->filename();

            $dataFormatter = $this->getDataFormatter($field, $filename);
            $data = $dataFormatter->formatData();
            $files = $this->constructFiles($entity, $data, $field, $settings, $basepath);

            $writer = $this->getWriter($entity, $data, $field, $settings);
            $success = $writer->write($files);

            if ((new Collection($success))->contains(false)) {
                return false;
            }

            $entity->set($field, $filename);
            $metaDataProcessor = $this->getGetMetaDataProcessor($files);
            $entity->set(Hash::get($settings, 'fields.dir', 'dir'), $basepath);
            $entity->set(Hash::get($settings, 'fields.size', 'size'), $metaDataProcessor->getSize());
            $entity->set(Hash::get($settings, 'fields.type', 'type'), $metaDataProcessor->getType());
```
Something like this.

Reply:
Maybe you could come with something better?I tried to think of a design pattern to apply, but I could only think of template or strategy but is pretty much like what we already have with the interfaces in place

Reply:
I meant in terms of an end user. How are you planning on using this/why should this go into the core?

Reply:
I'm using https://foliotek.github.io/Croppie/ so that the user can select his profile picture and the result is in base64.